zb-db/src/main/java/io/zeebe/db/impl/rocksdb/transaction/ColumnFamilyContext.java

```diff
@@ -42,35 +29,38 @@
   private final DirectBuffer valueViewBuffer = new UnsafeBuffer(0, 0);
 
   private final Queue<ExpandableArrayBuffer> prefixKeyBuffers;
+  private int keyLength;
 
-  DefaultDbContext(final ZeebeTransaction transaction) {
-    this.transaction = transaction;
+  ColumnFamilyContext() {
     prefixKeyBuffers = new ArrayDeque<>();
     prefixKeyBuffers.add(new ExpandableArrayBuffer());
     prefixKeyBuffers.add(new ExpandableArrayBuffer());
   }
 
-  @Override
-  public void writeKey(final DbKey key) {
-    key.write(keyBuffer, 0);
+  public void writeKey(final long keyPrefix, final DbKey key) {
+    keyLength = 0;
+    keyBuffer.putLong(0, keyPrefix, ZeebeDbConstants.ZB_DB_BYTE_ORDER);
```

Comment:
Can you elaborate on that?

Reply:
Every access to the db should write the keyPrefix which is now explicitly handled by the caller for this method. Since there is an instance of ColumnFamilyContext per columnfamily, it feels like keyPrefix can be a member of ColumnFamily and it implicitly writes the keyPrefix.

Reply:
Ah I think I understand now. You mean when we create the ColumnFamilyContext in the TransactionColumnFamily we can give it the type/key via the constructor and on calling writeKey etc it handles it internally (puts the type in front of the real key) right? Yeah we could do that. :)